modules/core/src/lib/api/load.js

```diff
@@ -13,20 +13,25 @@ export async function load(url, loaders, options) {
     loaders = null;
   }
 
-  // at this point, `url` could be already loaded binary data
-  let data = url;
+  // Normalize arguments
+  options = options || {};
+
+  // Select fetch function
+  const fetch = typeof options.fetch === 'function' ? options.fetch : fetchFile;
 
+  // at this point, `url` could be already loaded binary data
+  let data = url; 
   // url is a string, fetch the url
   if (typeof url === 'string') {
-    data = await fetchFile(url, options);
+    data = await fetch(url, options);
```

Comment:
I'm a little confused by passing `options` here. Since this is the same `options` object passed to `load`, it has more than just the fetch options? I would've expected a `fetch` key on the `options` object, like
```js
load(url, ImageLoader, {image: {type: 'imagebitmap'}, fetch: {mode: 'no-cors'}})
```
Then only the `.fetch` key would be passed to `fetch`

Reply:
Great observation. We do need to define the semantics of the fetch option. My current thinking is that `options.fetch` can either be a function or options to the default `fetchFile` functions. 

There is actually an old [RFC](https://github.com/visgl/loaders.gl/blob/master/dev-docs/RFCs/v2.0/fetch-option-rfc.md)  for this that needs updating. We should definitely go through that process before merging since this change also needs deck.gl sign-off.

Reply:
> `options.fetch` can either be a function or options to the default `fetchFile` functions

Yes that's what I had in mind as well. And `fetchFile` presumably passes those objects to `fetch` when in the browser.